### ISSUE TITLE
A namespace sharing the same name as a module with ModuleSuffix declared in the same file is not visible

### ISSUE THREAD

**Issue by mciccottidmg** [dotnet/fsharp#14799]
If there is a namespace which shares a name with a module, the namespace isn't visible in the project when the namespace and module are declared the same file (see https://github.com/dotnet/fsharp/issues/14798 for a similar bug when they're declared in separate files).

**Repro steps**

Provide the steps required to reproduce the problem:

1. Create a module and a namespace with the same name (the ModuleSuffix attribute is required for this to compile. Not sure if it should be)
```fsharp
namespace NamespaceModuleCollision

[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
module TopLevel =

    let test = 5

namespace NamespaceModuleCollision.TopLevel
    module Sub =

        let test2 = 5
```
2. Attempt to use the values in the modules in a different file in the same project
```fsharp
module library

let _ = NamespaceModuleCollision.TopLevel.test
let _ = NamespaceModuleCollision.TopLevel.Sub.test2 //error

open NamespaceModuleCollision.TopLevel
open NamespaceModuleCollision.TopLevel.Sub //error

let _ = test2 //obviously has an error since the module can't be opened
```
If possible attach a zip file with the repro case. This often makes it easier for others to reproduce.
The zip file should ideally represent the situation just before the call/step that is problematic.

**Expected behavior**

The values in both the modules are accessible.

**Actual behavior**
Compiler errors when attempting to access the values in the 'Sub' module:

error FS0039: The value, constructor, namespace or type 'Sub' is not defined.
error FS0039: The namespace 'Sub' is not defined.
error FS0039: The value or constructor 'test2' is not defined. Maybe you want one of the following:↔   test↔   Text

[library.zip](https://github.com/dotnet/fsharp/files/10818591/library.zip)

**Known workarounds**

1. Declare the module and namespace in different files, but note that this only fixes the problem within a project. The same issue exists when attempting to access the namespace in a different project. i.e. https://github.com/dotnet/fsharp/issues/14798

2. Change the name of the module or namespace.

**Comment by vzarytovskii** [comment]
Same here, I have tested it with different frameworks and SDKs, as old as netcoreapp3.1 and net472, and it still produces the same diagnostic, so if it's a regression, it's for sure not a recent one.

Not entirely sure it ever worked, @nojaf, @auduchinok, @dsyme Don Syme FTE thoughts? @mciccottidmg did it ever work for you?

**Comment by auduchinok** [comment]
> Create a module and a namespace with the same name (the ModuleSuffix attribute is required for this to compile. Not sure if it should be)

From the top of my head, I'd expect an error here, instead of or in addition to the errors in the second file.
I'm not sure if this case was ever supported.

**Comment by nojaf** [comment]
Nothing comes to mind here.

**Comment by auduchinok** [comment]
> Same here, I have tested it with different frameworks and SDKs, as old as netcoreapp3.1 and net472, and it still produces the same diagnostic, so if it's a regression, it's for sure not a recent one.

I think fsi could help to test it. Are namespaces allowed in `#load ...` files? If yes, it would allow to test it on different versions easily.

**Comment by mciccottidmg** [comment]
@vzarytovskii 
>@mciccottidmg did it ever work for you?

No, I'm guessing this never worked. The reason I assume it should work is that it _does_ work if you put the declarations in different files. Whether the declarations are in the same file or not seems like it shouldn't matter.

**Comment by mciccottidmg** [comment]
I would guess this issue has the same cause as https://github.com/dotnet/fsharp/issues/3565 or at least closely related to it.

**Comment by mciccottidmg** [comment]
The reason I want this, is I have a module with some general functions in it, and then more specific functionality in a sub-module, but that more specific functionality is rather lengthy (1300 lines) and I want to put the different sub-modules in different files. If I can put TopLevel and TopLevel->Sub1, TopLevel->Sub2 etc in different files, that helps a lot with code organization.

**Comment by NinoFloris** [comment]
@vzarytovskii this one should still be a bug no? It's the same file in the same project not being patched up correctly?

**Comment by vzarytovskii** [comment]
> @vzarytovskii Vlad Zarytovskii FTE this one should still be a bug no? It's the same file in the same project not being patched up correctly?

I haven't found a verision of compiler where it works, just yet.

**Comment by mciccottidmg** [comment]
Minor correction: it appears that it's the _order_ of the declarations which matters, not being in the same file or not.

**Comment by xperiandri** [comment]
That is true
